### PR TITLE
Clean up Cython tests

### PR DESCRIPTION
**Pull request by mdickinson**
Somewhat experimental PR to clean up the Cython tests:

- Cythonize with "language_level=3" to suppress warnings from Cython in the test output. (There's still an unreachable code warning from the compiler; there doesn't appear to be much we can do about that.)
- Factor out common code.
- Don't test whether Cython is usable in order to decide whether to skip tests: assume that if Cython is installed, then it's usable. This removes an import-time side-effect and simplifies the code.
- Remove a test that was designed solely to record an existing failing case; I've turned that into an issue instead (see #554).

### CODE REVIEW

**Comment by mdickinson** [comment]
Happy everywhere except Windows / Python 2.7. Failures below.

```
======================================================================
ERROR: test_basic_events (traits.tests.test_cythonized_traits.CythonizedTraitsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 49, in test_basic_events
    obj = self.execute(code)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 205, in execute
    language_level="3",
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Shadow.py", line 133, in inline
    return _cython_inline(f, *args, **kwds)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Build\Inline.py", line 250, in cython_inline
    build_extension.run()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self.build_extension(ext)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 299, in query_vcvarsall
    raise ValueError(str(list(result.keys())))
ValueError: [u'path']
======================================================================
ERROR: test_on_trait_aliasing (traits.tests.test_cythonized_traits.CythonizedTraitsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 164, in test_on_trait_aliasing
    obj = self.execute(code)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 205, in execute
    language_level="3",
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Shadow.py", line 133, in inline
    return _cython_inline(f, *args, **kwds)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Build\Inline.py", line 250, in cython_inline
    build_extension.run()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self.build_extension(ext)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 299, in query_vcvarsall
    raise ValueError(str(list(result.keys())))
ValueError: [u'path']
======================================================================
ERROR: test_on_trait_aliasing_different_scope (traits.tests.test_cythonized_traits.CythonizedTraitsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 190, in test_on_trait_aliasing_different_scope
    obj = self.execute(code)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 205, in execute
    language_level="3",
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Shadow.py", line 133, in inline
    return _cython_inline(f, *args, **kwds)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Build\Inline.py", line 250, in cython_inline
    build_extension.run()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self.build_extension(ext)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 299, in query_vcvarsall
    raise ValueError(str(list(result.keys())))
ValueError: [u'path']
======================================================================
ERROR: test_on_trait_on_trait_change_decorator (traits.tests.test_cythonized_traits.CythonizedTraitsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 90, in test_on_trait_on_trait_change_decorator
    obj = self.execute(code)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 205, in execute
    language_level="3",
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Shadow.py", line 133, in inline
    return _cython_inline(f, *args, **kwds)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Build\Inline.py", line 250, in cython_inline
    build_extension.run()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self.build_extension(ext)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 299, in query_vcvarsall
    raise ValueError(str(list(result.keys())))
ValueError: [u'path']
======================================================================
ERROR: test_on_trait_properties (traits.tests.test_cythonized_traits.CythonizedTraitsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 112, in test_on_trait_properties
    obj = self.execute(code)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 205, in execute
    language_level="3",
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Shadow.py", line 133, in inline
    return _cython_inline(f, *args, **kwds)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Build\Inline.py", line 250, in cython_inline
    build_extension.run()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self.build_extension(ext)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 299, in query_vcvarsall
    raise ValueError(str(list(result.keys())))
ValueError: [u'path']
======================================================================
ERROR: test_on_trait_properties_with_standard_getter (traits.tests.test_cythonized_traits.CythonizedTraitsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 136, in test_on_trait_properties_with_standard_getter
    obj = self.execute(code)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 205, in execute
    language_level="3",
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Shadow.py", line 133, in inline
    return _cython_inline(f, *args, **kwds)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Build\Inline.py", line 250, in cython_inline
    build_extension.run()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self.build_extension(ext)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 299, in query_vcvarsall
    raise ValueError(str(list(result.keys())))
ValueError: [u'path']
======================================================================
ERROR: test_on_trait_static_handlers (traits.tests.test_cythonized_traits.CythonizedTraitsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 68, in test_on_trait_static_handlers
    obj = self.execute(code)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 205, in execute
    language_level="3",
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Shadow.py", line 133, in inline
    return _cython_inline(f, *args, **kwds)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Build\Inline.py", line 250, in cython_inline
    build_extension.run()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self.build_extension(ext)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 299, in query_vcvarsall
    raise ValueError(str(list(result.keys())))
ValueError: [u'path']
======================================================================
ERROR: test_simple_default_methods (traits.tests.test_cythonized_traits.CythonizedTraitsTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 35, in test_simple_default_methods
    obj = self.execute(code)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\traits\tests\test_cythonized_traits.py", line 205, in execute
    language_level="3",
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Shadow.py", line 133, in inline
    return _cython_inline(f, *args, **kwds)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\site-packages\Cython\Build\Inline.py", line 250, in cython_inline
    build_extension.run()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 340, in run
    self.build_extensions()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 449, in build_extensions
    self.build_extension(ext)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\command\build_ext.py", line 499, in build_extension
    depends=ext.depends)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 473, in compile
    self.initialize()
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 383, in initialize
    vc_env = query_vcvarsall(VERSION, plat_spec)
  File "C:\Users\appveyor\.edm\envs\traits-test-2.7\lib\distutils\msvc9compiler.py", line 299, in query_vcvarsall
    raise ValueError(str(list(result.keys())))
ValueError: [u'path']
```

**Comment by mdickinson** [comment]
This is puzzling: we clearly _have_ a working C compiler setup and the ability to build Python extensions, else we wouldn't have been able to build the `ctraits` extension at the heart of Traits.

@itziakos Any idea what might be going wrong here?

I'll see if I can reproduce and debug locally, but I'm not going to sink a lot of time into this: since the problem is Python 2.7 only, the simplest path to a passing CI run may be simply not to include Cython in the test environment for Python 2.7 / Windows.

**Comment by itziakos** [comment]
> 
> 
> This is puzzling: we clearly _have_ a working C compiler setup and the ability to build Python extensions, else we wouldn't have been able to build the `ctraits` extension at the heart of Traits.
> 
> @itziakos Any idea what might be going wrong here?
> 
> I'll see if I can reproduce and debug locally, but I'm not going to sink a lot of time into this: since the problem is Python 2.7 only, the simplest path to a passing CI run may be simply not to include Cython in the test environment for Python 2.7 / Windows.

Hi Mark, I was not able to reproduce this issue. I have not seen this before.

**Comment by mdickinson** [comment]
Opened #557 for the failure. We'll go back to skipping these tests on the Windows / Python 2.7 combination.

File: traits/tests/test_cythonized_traits.py [hunk 1]
```diff
@@ -153,25 +124,18 @@ def _get_name_len(self):
...
         # Assert dependency works
         obj.name = "Bob"
         self.assertEqual(obj.name_len, len(obj.name))
 
-    @unittest.skipIf(SKIP_TEST, "Missing Cython and/or compiler")
     def test_on_trait_properties_with_standard_getter(self):
 
         code = """
 from traits.api import HasTraits, Str, Int, Property
+from traits.api import Unicode
```
Comment (mdickinson) [inline traits/tests/test_cythonized_traits.py]: Unrelated change. Will revert.

**Comment by mdickinson** [comment]
Self-merging.
